pymc_marketing/product_incrementality/mv_its.py

```diff
@@ -0,0 +1,403 @@
+    ):
+        """Return a PyMC model for a multivariate interrupted time series analysis."""
+
+        if not background_sales.index.equals(innovation_sales.index):
+            raise ValueError(
+                "Index of background_sales and innovation_sales must match."
+            )
+
+        # note: type hints for coords required for mypi to not get confused
+        coords: dict[str, list[str]] = {
+            "background_product": list(background_sales.columns),
+            "time": list(background_sales.index.values),
+            "all_sources": [
+                *list(background_sales.columns),
+                "new",
+            ],  # for non-saturated market only
+            # "all_sources": list(background_sales.columns)
+            # + ["new"],  # for non-saturated market only
+        }
+
+        with pm.Model(coords=coords) as model:
+            # data
+            _background_sales = pm.Data(
+                "background_sales",
+                background_sales.values,
+                dims=("time", "background_product"),
+            )
+            innovation_sales = pm.Data(
+                "innovation_sales", innovation_sales.values, dims=("time",)
+            )
+
+            # priors
+            intercept = pm.Normal(
+                "intercept",
+                mu=pm.math.mean(background_sales[:treatment_time], axis=0),
+                sigma=np.std(background_sales[:treatment_time], axis=0),
+                # sigma=20,
+                dims="background_product",
+            )
+
+            sigma = pm.HalfNormal(
+                "background_product_sigma", sigma=10, dims="background_product"
+            )
+
+            if market_saturated:
+                """We assume the market is saturated. The sum of the beta's will be 1.
+                This means that the reduction in sales of existing products will equal
+                the increase in sales of the new product, such that the total sales
+                remain constant."""
+                alpha = np.full(len(coords["background_product"]), alpha_background)
+                beta = pm.Dirichlet("beta", a=alpha, dims="background_product")
+            else:
+                """We assume the market is not saturated. The sum of the beta's will be
+                less than 1. This means that the reduction in sales of existing products
+                will be less than the increase in sales of the new product."""
+                alpha_all = np.full(len(coords["all_sources"]), alpha_background)
+                beta_all = pm.Dirichlet("beta_all", a=alpha_all, dims="all_sources")
+                beta = pm.Deterministic(
+                    "beta", beta_all[:-1], dims="background_product"
+                )
+                pm.Deterministic("new sales", beta_all[-1])
+
+            # expectation
+            mu = pm.Deterministic(
+                "mu",
+                intercept[None, :] - innovation_sales[:, None] * beta[None, :],
```

Comment:
What are the innovation sales and the background sales supposed to mean? Do you expect this to be a general enough relationship? Is the term innovation supposed to distinguish between a set of products that you consider the treatment (or target, I don’t remember the proper name) for the interrupted time series, and another set of products that aren’t?